multicluster/controllers/multicluster/leader/labelidentity_export_controller_test.go

```diff
@@ -185,6 +191,61 @@ func TestLabelIdentityResourceExportReconclie(t *testing.T) {
 	}
 }
 
+func TestConcurrentProcessLabelForResourceImport(t *testing.T) {
+	existingResImp := &mcsv1alpha1.ResourceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: common.LeaderNamespace,
+			Name:      labelHash3,
+		},
+		Spec: mcsv1alpha1.ResourceImportSpec{
+			ClusterIDs: []string{common.LocalClusterID},
+			LabelIdentity: &mcsv1alpha1.LabelIdentitySpec{
+				Label: normalizedLabel3,
+				ID:    3,
+			},
+		},
+	}
+	existResImpList := &mcsv1alpha1.ResourceImportList{
+		Items: []mcsv1alpha1.ResourceImport{*labelIdentityResImp, *existingResImp},
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	fakeClient := fake.NewClientBuilder().WithScheme(common.TestScheme).WithLists(existResImpList).Build()
+	r := NewLabelIdentityExportReconciler(fakeClient, common.TestScheme, common.LeaderNamespace)
+	r.hashToLabels = map[string]string{
+		labelHash:  normalizedLabel,
+		labelHash2: normalizedLabel2,
+	}
+	// Mock the state of the reconciler where id 3 was allocated for labelHash3
+	r.labelsToID.Store(labelHash3, uint32(3))
+	r.allocator.setAllocated(3)
+	// The three keys should be drawn out of the queue almost concurrently after the worker starts
+	r.labelQueue.Add(labelHash)
+	r.labelQueue.Add(labelHash2)
+	r.labelQueue.Add(labelHash3)
+	go r.Run(stopCh)
+
+	// Wait until all items in the label queue are processed
+	wait.PollWithContext(common.TestCtx, 10*time.Millisecond, time.Second, func(context.Context) (done bool, err error) {
+		if r.labelQueue.Len() == 0 {
```

Comment:
This only means the workers have received the keys from the queue but doesn't mean the workers have finished their works. There could be two ways to perform the test you want:
1. Run the sync function concurrently directly and wait for them explicitly:
```
for _, labelHash := range []string{labelHash, labelHash2, labelHash3} {
    wg.Add(1)
    go func() {
        defer wg.Done()
        r.syncLabelResourceImport(labelHash)
    }
}
wg.Wait()
...
```
2. Use something like assert.Eventually to check if the desired result can be reached in a specified time:
```
assert.Eventually(func() bool {
    actLabelIdentityResImpList := &mcsv1alpha1.ResourceImportList{}
    fakeClient.List(common.TestCtx, actLabelIdentityResImpList)
    if len(actLabelIdentityResImpList.Items) != 2 {
        return false
    }
    ...
}, time.Millisecond*10, time.Second, ...)
```